Source/DafnyCore/Plugins/IExecutableBackend.cs

```diff
@@ -112,6 +112,9 @@ public abstract class IExecutableBackend {
   protected ErrorReporter? Reporter;
   protected ReadOnlyCollection<string>? OtherFileNames;
 
+  // The following lists are the Options supported by the backend.
+  public virtual IEnumerable<Option<string>> SupportedOptions => new List<Option<string>>();
```

Comment:
Having cross-backend options is a fine concept in general btw, even if `--go-module-name` shouldn't be a cross-backend option.

Although this approach means that if you do `dafny translate java --some-option-java-doesnt-support` you'll get an argument parsing error rather than the opportunity to have a clean "Java doesn't support the --some-option-java-doesnt-support option" error.

Again if YAGNI it might be simpler to just remove this for now.

Reply:
Looked at the current state of the codebase and realized there isn't already a place to stick options that are specific to a backend, so this is a nice improvement so that every backend doesn't end up having to override `GetCommand()` in a similar way.

But if you do keep this you should also update `CppBackend.GetCommand()` accordingly, since it overrides this version.

Reply:
Yup, this was done to provide a hook for other backends to implement their custom `Options`. I've fixed `CppBackend.GetCommand()` to add it's custom description to the `TragetId` Option.